src/github.com/getlantern/flashlight/client/balancer.go

```diff
@@ -14,9 +18,22 @@ func (client *Client) initBalancer(cfg *ClientConfig) *balancer.Balancer {
 	dialers := make([]*balancer.Dialer, 0, len(cfg.FrontedServers)+len(cfg.ChainedServers))
 
 	log.Debugf("Adding %d domain fronted servers", len(cfg.FrontedServers))
+	var highestQOSFrontedDialer fronted.Dialer
+	highestQOS := math.MinInt32
 	for _, s := range cfg.FrontedServers {
-		dialer := s.dialer(cfg.MasqueradeSets)
+		fd, dialer := s.dialer(cfg.MasqueradeSets)
 		dialers = append(dialers, dialer)
+		if dialer.QOS > highestQOS {
+			highestQOSFrontedDialer = fd
+		}
+	}
+
+	if highestQOSFrontedDialer != nil {
+		// This function will configure the HTTP client the geo package is going to
+		// use.
+		geolookup.SetHTTPClient(highestQOSFrontedDialer.DirectHttpClient())
```

Comment:
Because this SetHTTPClient() changes a variable in another package, I don't quite like working with global values that way because it's difficult to know if the current value is different from the original.